stackslib/src/chainstate/nakamoto/tenure.rs

```diff
@@ -764,84 +775,34 @@ impl NakamotoChainState {
                     );
                     return Ok(None);
                 }
-                if tenure_payload.burn_view_consensus_hash
```

Comment:
I'm beginning to think that a lot of the assumptions that went into this module are wrong.  In particular, this module assumes that the tenure history does not fork unless the Bitcoin history also forks, but it seems that this is now a design requirement?  This assumption made sense while we were also assuming that the signer would never permit a Stacks fork, but it appears that this is actually unavoidable.

I think some substantial refactoring is in order.

Reply:
(This doesn't have to be done in this PR; I'll create a separate ticket once I can spec out the work)